python/tank/platform/engine.py

```diff
@@ -1832,14 +1836,69 @@ def start_engine(engine_name, tk, context):
     Raises TankEngineInitError if an engine could not be started
     for the passed context.
     """
+    return _start_engine(engine_name, tk, None, context)
+
+
+def _restart_engine(new_context):
+
+    engine = current_engine()
+    try:
+        # Track some of the current state before restarting the engine.
+        old_context = engine.context
+
+        # Restart the engine. If we were given a new context to use,
+        # use it, otherwise restart using the same context as before.
+        new_context = new_context or engine.context
```

Comment:
So in the case of a normal engine restart, the context switching hooks will fire, but the old and new context is the same? Does that make sense? Seems to me in that case, the context isn't actually changing....